app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java

```diff
@@ -88,7 +93,15 @@ public static Vault fromJson(JSONObject obj) throws VaultException {
         return vault;
     }
 
-    public void migrateOldGroup(VaultEntry entry) {
+    private void setGroupsMigrated() {
+        _groupsMigrated = true;
+    }
+
+    public boolean getGroupsMigrated() {
+        return _groupsMigrated;
+    }
+
+    public boolean migrateOldGroup(VaultEntry entry) {
```

Comment:
Perhaps 'needsGroupMigration' is a better name for this function.

Reply:
This function doesn't just check whether a group needs to be migrated, it also actually performs the migration.